engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/LazyConstructorTest.java

```diff
@@ -0,0 +1,107 @@
+package org.enso.interpreter.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.ByteArrayOutputStream;
+import org.enso.polyglot.MethodNames;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.PolyglotException;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class LazyConstructorTest extends TestBase {
+  private static final ByteArrayOutputStream out = new ByteArrayOutputStream();
+  private static Context ctx;
+
+  public LazyConstructorTest() {}
+
+  @BeforeClass
+  public static void prepareCtx() {
+    ctx = createDefaultContext(out);
+  }
+
+  @Before
+  public void resetOut() {
+    out.reset();
+  }
+
+  @AfterClass
+  public static void disposeCtx() {
+    ctx.close();
+  }
+
+  @Test
+  public void lazyConstructorWithNoArgument() {
+    try {
+      var create =
+          ctx.eval(
+                  "enso",
+                  """
+              type N
+                  False
+
+                  materialize v:N = v.to_text
+
+              create n = N.materialize (~False)
+              """)
+              .invokeMember(MethodNames.Module.EVAL_EXPRESSION, "create");
+      assertTrue("Can evaluate", create.canExecute());
+      assertEquals("False", create.execute(42).asString());
+
+    } catch (PolyglotException e) {
+      fail(e.getMessage() + " for \n" + out.toString());
+    }
+  }
+
+  @Test
+  public void lazyConstructorWithSingleArg() {
+    try {
+      var create =
+          ctx.eval(
+                  "enso",
+                  """
+              type M
+                  Construct value
+
+                  materialize v:M = v.value
+
+              create n = M.materialize (~Construct n)
```

Comment:
I'm not 100% convinced about the assertion
```java
assert this.args == NONE;
```
in the code below.

So to ensure all is working OK, I'd like to propose the following additional tests:
1. Have a constructor with 3 arguments and have `M.materialize (~Construct 1 2 3)` to ensure many arguments work ok.

2. What happens if I have an unresolved constructor and _call_ it? e.g.

```
type M
    Construct a b c
    materialize v:M = v.a+v.b+v.c

foo =
	x = ~Construct 1
    y = x 2
    z = x 3
    M.materialize z
```

I'm not sure what even is the right behaviour here. I guess either:
a) treat the constructor like a function and allow it to gather arguments, then materialize it (so `foo` returns `6`),
b) OR do not allow currying the constructor (so `x 2` fails with `Not_Invokable` error).

It feels like for the language to be more consistent, (a) seems to be better. But I'm not sure if it does not open some other issues.

Reply:
I also think we should have a bit more test coverage.

1. What happens if upon resolution it turns out the constructor had too many / too few arguments?
2. Let's ensure that this also works correctly if the constructor has default arguments. E.g. a case `M.materialize ~Construct` where constructor had 1 defaulted argument, and `M.materialize (~Construct 1)` where the constructor had 2 arguments, the latter (or both?) being defaulted.
3. What if I use named (and out of order) arguments for my constructor? e.g. `M.materialize (~Constructor b=1 a=2)`? Normal constructors handle this case well, so I guess suspended ones should too, but I'm not convinced the current implementation handles this (unless I'm missing some hidden magic, but still that should be tested anyway).

Especially with (2) and (3) are important because that's what happens in the IDE. E.g. when I select a `Aggregate_Column.Sum` in `Table.aggregate`, the `new_name` (or after rename, `as`) argument is defaulted. If all arguments are defaulted and I only override the value of e.g. 3rd one, the IDE uses named parameters to indicate that. So I think the suspended constructors need to support this.

Of course this may not necessarily have to be done in this PR, but if not we probably need to have a ticket to be able to plan such additional work.

Reply:
Thank you Radek for great comments.
- two (and thus multi) argument constructor works as demonstrated in 0342ce3
- ...